Tests/Realm.Tests/Database/CollectionTests.cs

```diff
@@ -247,6 +247,37 @@ public void ObjectList_WhenEnumeratingAndRemovingFromList_ShouldBeStable()
             Assert.That(items, Is.EquivalentTo(Enumerable.Range(0, 10)));
         }
 
+        [Test]
+        public void ObjectList_WhenEnumeratingFrozenList_ShouldBeStable()
+        {
+            var container = new ContainerObject();
+            _realm.Write(() =>
+            {
+                _realm.Add(container);
+            });
+
+            var j = 10;
+            var frozenList = container.Items.Freeze();
+
+            TestStableIteration(
+                i => container.Items.Add(new IntPropertyObject { Int = i }),
+                () => frozenList = container.Items.Freeze(),
+                item =>
+                {
+                    // Just remove last from container since item from
+                    // frozen version is not equivalent.
+                    j -= 1;
+                    container.Items.RemoveAt(j);
```

Comment:
I had trouble coming up with a meaningful test for this since I don't think you are meant to be removing things from frozen collections... so this is a somewhat weird experiment that mixes frozen collection with original collection (not that the two are affecting each other anyways).